diff --git a/usr/local/share/perl/5.10.1/Net/Server/Proto/SSL.pm b/usr/local/share/perl/5.10.1/Net/Server/Proto/SSL.pm.jet
index b39b2ea..45dc282 100644
--- a/usr/local/share/perl/5.10.1/Net/Server/Proto/SSL.pm
+++ b/usr/local/share/perl/5.10.1/Net/Server/Proto/SSL.pm.jet
@@ -76,7 +76,12 @@ sub object {
 
   for my $key (@ssl_args) {
     my $val = defined($prop->{$key}) ? $prop->{$key} : $server->can($key) ? $server->$key($host, $port, 'SSL') : undef;
-    $sock->$key($val);
+    if( $sock->can($key) ) {
+        $sock->$key($val);
+
+    } else {
+        $sock->{$key} = $val;
+    }
   }
 
   return $sock;
